system/gsettings.py

```diff
@@ -0,0 +1,248 @@
+---
+module: gsettings
+short_description: Module to manage dconf settings.
+description:
+    - This module provides a python equivalent of the gsettings command-line tool
+    - for use in ansible.
+version_added: "1.9"
+notes:
+    - become and become_user must be used to specify the target configuration set.
+requirements: [ "PyGObject"]
+options:
+    settings:
+        description:
+            - A list of schema, key, value triplets to update.
+        required: true
+        default: None
+        version_added: 1.9
+author: Adrian Wood (@ag-wood)
+'''
+
+EXAMPLES = '''
+# Desktop settings for Ubuntu Unity Desktop:
+- hosts: all
+  gather_facts: no
+  tasks:
+    - name: Change Desktop configuration
+    become: yes
+    become_user: janedoe
+    gsettings:
+      settings:
+      # Ensure lock when screensaver activates
+      - schema: org.gnome.desktop.screensaver
+        key: lock-enabled
+        value: true
+      # Clean up the 'Dash' search widget
+      - schema: com.canonical.Unity.Lenses
+        key: disabled-scopes
+        value: ['more_suggestions-amazon.scope', 'more_suggestions-u1ms.scope',
+                'more_suggestions-populartracks.scope', 'music-musicstore.scope',
+                'more_suggestions-ebay.scope', 'more_suggestions-ubuntushop.scope',
+                'more_suggestions-skimlinks.scope']
+      # Power button invokes shutdown.
+      - schema: org.gnome.settings-daemon.plugins.power
+        key: button-power
+        value: 'shutdown'
+      # Desktop colours and wallpaper.
+      - schema: org.gnome.desktop.background
+        key: primary-color
+        value: '#000000'
+      - schema: org.gnome.desktop.background
+        key: secondary-color
+        value: '#000000'
+      - schema: org.gnome.desktop.background
+        key: picture-uri
+        value: 'file:///usr/share/backgrounds/Beach_by_Renato_Giordanelli.jpg'
+'''
+
+# Standard python libraries
+import os
+import pwd
+import time
+import signal
+import subprocess
+
+try:
+    from gi.repository import Gio, GLib
+except ImportError:
+    HAS_GIO = False
+else:
+    HAS_GIO = True
+
+ENV_DBUS = 'DBUS_SESSION_BUS_ADDRESS'
+SESSION_MANAGERS = [
+    'gnome-session', 'mate-session', 'xfce4-session',
+    'cinnamon-session', 'icewm-session', 'openbox-session']
+
+
+class Gsettings(object):
+    def __init__(self):
+        def proc_name(pid):
+            """Return process name of process pid"""
+            try:
+                for attr in open("/proc/{0}/status".format(pid), 'r'):
+                    if attr.startswith('Name:'):
+                        return attr.split(':')[1].strip()
+                return None
+            except:
+                return None
+
+        def proc_owner(pid):
+            """Return username of UID of process pid"""
+            UID = 1
+            for ln in open('/proc/{0}/status'.format(pid), 'r'):
+                if ln.startswith('Uid:'):
+                    return int(ln.split()[UID])
```

Comment:
Uh.  I don't see why you use a named constant here but no such thing in `proc_name()`.

I don't think named constants help, you're just taking the 2nd space-separated field, so `ln.split()[1]` is fine.